### ISSUE TITLE
Few ABIs are wrong

### ISSUE THREAD

**Issue by the-exodus** [Gallopsled/pwntools#1394]
The SysV amd64 ABI specifies the stack has to be 16-byte aligned on function calls, while the code says `linux_amd64  = ABI('rsp', ['rdi','rsi','rdx','rcx','r8','r9'], 8, 0)`

On linux, gcc made their own modification to the i386 ABI, requiring 16-byte alignment. The regular SysV abi still has the 8-byte alignment though.

Because of this, calls to functions using SSE (system() in glibc for example), generated with ROP::call() may end up failing depending on where it's ended up.

I would fix this and submit a PR, but I'm unsure if setting the right arg_alignment is all that is needed, or if it could cause any side effects.

**Comment by Arusekk** [comment]
Have you tested it locally? If you wonder how to do this, there's a [TESTING.md](/Gallopsled/pwntools/blob/dev/TESTING.md) file as introduction.

After you find your tests ok, or encounter a harder problem, please submit a PR and we'll see the tests automatically run for this.

**Comment by the-exodus** [comment]
@Arusekk Well, I tested it on the stable branch (it's what I was using up until now), only to realize it has no effect whatsoever. The `arg_align` and `stack_minimum` doesn't seem to be referenced anywhere in the code-base and the stack will only be adjusted for Call() with stack arguments.

Then I realized ROP::call() was behaving a bit odd overall and switched to the 4.1 branch. I didn't get around to changing the ABI-specs there, as Call-entries in the chain didn't looked like they were fully resolving and I didn't have time to look into it.

I will poke at it a bit in the morning, and hopefully you have good test-coverage :P
